uswds/uswds-site
# USWDS-Site - Ruby: Use `3.3.4`

# Summary

Sets ruby version to `3.3.4` and updates relevant gems.

## Related issue

Closes #1759

Updating in an effort to resolve the following dependabot alerts: 
- https://github.com/uswds/uswds-site/security/dependabot/71
- https://github.com/uswds/uswds-site/security/dependabot/70
- https://github.com/uswds/uswds-site/security/dependabot/69

## Preview link

Preview link →

## Problem statement

Ruby `3.1.x` versions have stopped receiving updates as of 2024-04-01. Additionally, REXML was causing 

## Solution

Update ruby version to `3.3.4`.

## Major changes

**Replaced `File.exsits?` with `File.exist?`** 
Starting in **Ruby 3.2.0** the `exists?` (pluralized) alias for `exist?` seems to has been removed.

## Testing and review

On your local machine, install and use Ruby `3.3.4`

1. Pull down and checkout this branch
2. Run `rvm install 3.3.4`.
  - If you have an install installing you may need to install and specify the OpenSSL version
  - `rvm install 3.3.4 -- --with-openssl-dir=$(brew --prefix openssl@1.1)`
3. Run `rvm use 3.3.4`
4. Confirm the correct ruby version is being used by running `ruby -v`
5. Run `gem install bundler` to confirm that the correct bundler version is used
6. Run `bundle install` to install all gems and confirm there are no errors or additional gems
7. Run `npm start` and confirm the site builds without errors

>[!IMPORTANT]
>Any additional PLATFORMS added to the gemfile should captured in #2577 and be ignored in this PR

### Testing checklist

- [ ] Gems install without error.
- [ ] Jekyll start and build commands run without error.
- [ ] No errors in `ci/circleci: install_ruby_deps`

## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| Ruby |     `3.1.2`      |   `3.3.4`   |
